src/AdoPat/PatClient.cs

```diff
@@ -32,12 +32,7 @@ public PatClient(ITokensHttpClientWrapper client)
             this.client = client;

```

Comment:
I was revisiting the TokensHttpClientWrapper implementation and I noticed that you create an http client in the constructor and store it for all methods to reuse. I have seen this been suggested as an anti-pattern because the same http client is reused for multiple requests. Instead, we could get a client per request and allow VSSConnection to cache the http clients depending how many are being created. This is a similar pattern where we should use HttpClientFactory to create http clients instead of storing http clients in our own class. (good read - https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests)

Azure Devops samples use this pattern -
https://github.com/microsoft/azure-devops-dotnet-samples/blob/main/ClientLibrary/Samples/WorkItemTracking/WorkItemsSample.cs

Reply:
I don't create an HTTP client in the constructor, it's injected. But I _do_ store it for all methods to use, yes. I did that on purpose in order to re-use the connection for the lifetime of this program, which should be **very** short.

The "`client`" here is not directly an [`System.Net.Http.HttpClient`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-7.0), but an instance of [`TokensHttpClientWrapper`](https://github.com/AzureAD/microsoft-authentication-cli/blob/93e2778c0eefdec3d9c9f7665b422c73b591a930/src/AdoPat/TokensHttpClientWrapper.cs), which itself is a thin wrapper around [`TokensHttpClient`](https://github.com/AzureAD/microsoft-authentication-cli/blob/93e2778c0eefdec3d9c9f7665b422c73b591a930/src/AdoPat/TokensHttpClientWrapper.cs#L26) by using `VssConnection.GetClient<TokensHttpClient>()`.

That client is then injected into the `PatManager` and used where necessary, which I _think_ matches the ["Typed Client"](https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests#implement-your-typed-client-classes-that-use-the-injected-and-configured-httpclient) pattern demonstrated in the article you linked.


Also, from reading that link you provided I see
> Therefore, HttpClient is intended to be instantiated once and reused throughout the life of an application.
> ...
> This can be a good solution for short-lived console apps or similar, that run a few times a day.

That's exactly my intention here. This whole thing should conclude in a couple hundred milliseconds at most and I think it should be reasonable to use a single socket during that time.

Reply:
I used a similar pattern in a code where many concurrent http requests were made and I ran into a network exhaustion problem. However, this shouldn't be a problem here.

Reply:
We may want to use `HttpClientFactory` if we have similar code in the library since consumer can have high scale scenarios.